src/project-css/app/Http/Controllers/TableController.php

```diff
@@ -548,50 +651,72 @@ public function worker(Request $request) {
         $curLine = str_replace('"', "", $curLine);
 
         // Tokenize the line
-        $tkns = $this->tknze($curLine);
+        $tkns = $this->tknze($curLine, $delimiter);
 
         // Validate the tokens and filter them
         $tkns = $this->fltrTkns($tkns);
 
-        // Size of both column array and data should be same
         // Count of tokens
         $orgCount = count($clmnLst) - 1;
 
-        if (count($tkns) == $orgCount) {
-          // Declae an array
-          $curArry = array();
+        $tknCount = count($tkns);
 
-          // Compact them into one array with utf8 encoding
-          for ($i = 0; $i<$orgCount; $i++) {
-            // added iconv to strip out invalid characters
-            $curArry[ strval($clmnLst[ $i ]) ] = utf8_encode($tkns[ $i ]);
-          }
+        $valid_tkn_count = ($tknCount == $orgCount);
 
-          // remove extra characters replacing them with spaces
-          // also remove .. that is in the filenames
-          $cleanString = preg_replace('/[^A-Za-z0-9._ ]/', ' ', str_replace('..', '', $curLine));
+        // if lastErrRow is the previous row try to combine the lines
+        if (($valid_tkn_count == false) && ($lastErrRow == $prcssd-1) && ($savedTkns != NULL)) {
+            $tkns = $this->mergeLines($savedTkns, $tkns);
+            $tknCount = count($tkns);
+            $valid_tkn_count = ($tknCount == $orgCount);
 
-          // remove extra spaces and make string all lower case
-          $cleanString = strtolower(preg_replace('/\s+/', ' ', $cleanString));
+            // clear last error since we did a merge
+            $lastErrRow = NULL;
+            $savedTkns = NULL;
+        }
 
-          // remove duplicate keywords in the srchindex
-          $srchArr = explode(" ", $cleanString);
-          //$srchArr = array_unique( $srchArr );
+        try {
+          // throw a exception if we do not have a valid token count
+          if ($valid_tkn_count) {
+            // Declae an array
+            $curArry = array();
```

Comment:
consider different way of handling, either an if or using the try catch in a more appropriate way.